app/scripts/directives/notifications/notificationDrawerWrapper.js

```diff
@@ -0,0 +1,288 @@
+          // convert the map into a sorted array
+          var sortedGroups = _.sortBy(groupsMap, function(group) {
+            return group.heading;
+          });
+          // and sort the notifications under each one
+          _.each(sortedGroups, function(group) {
+            group.notifications = sortNotifications(group.notifications);
+            group.counts = countUnreadNotificationsForGroup(group);
+          });
+          return sortedGroups;
+        };
+
+        var formatFilteredEvents = function(eventMap) {
+          var filtered = {};
+          _.each(eventMap, function(event) {
+            if(EventsService.isImportantEvent(event)) {
+              ensureProjectGroupExists(filtered, event.metadata.namespace);
+              filtered[event.metadata.namespace].notifications.push({
+                unread:  !EventsService.isRead(event),
+                message: event.message,
+                lastTimestamp: event.lastTimestamp,
+                metadata: event.metadata,
+                involvedObject: event.involvedObject,
+                name: event.involvedObject.name,
+                kind: event.involvedObject.kind,
+                namespace: event.involvedObject.namespace,
+                status: event.type,
+                reason: event.reason,
+                actions: null
+              });
+            }
+          });
+          return filtered;
+        };
+
+        var eventWatchCallback = function(eventData) {
+          var eventsByName = eventData.by('metadata.name');
+
+          notificationGroupsMap = formatFilteredEvents(eventsByName);
+          notificationGroups = sortNotificationGroups(notificationGroupsMap);
+
+          $rootScope.$apply(function () {
+            // NOTE: we are currently only showing one project in the drawer at a
+            // time. If we go back to multiple projects, we can eliminate the filter here
+            // and just pass the whole array as notificationGroups.
+            // if we do, we will have to handle group.open to keep track of what the
+            // user is viewing at the time & indicate to the user that the non-active
+            // project is "asleep"/not being watched.
+            drawer.notificationGroups = _.filter(notificationGroups, function(group) {
+              return group.project.metadata.name === $routeParams.project;
+            });
+          });
+        };
+
+        // TODO: atm these are not being added to the notificationGroups
+        var notificationWatchCallback = function(event, notification) {
+          if(!notification.lastTimestamp) {
+            notification.lastTimestamp = new Date();
+          }
+          eventsFromNotifications.push(notification);
+        };
+
+        var deregisterRootScopeWatches = function() {
+          _.each(rootScopeWatches, function(deregister) {
+            deregister();
+          });
+        };
+
+        var toggleDrawerVisibility = function(projectName) {
```

Comment:
This makes it sound like opening and closing the drawer. Suggest a different name. It's not actually toggling, just hiding sometimes.